src/github.com/getlantern/flashlight/client/balancer.go

```diff
@@ -55,26 +44,15 @@ func (client *Client) initBalancer(cfg *ClientConfig) {
 	}
 
 	bal := balancer.New(dialers...)
-
-	if client.balInitialized {
-		log.Trace("Draining balancer channel")
-		old := <-client.balCh
+	oldBal, ok := client.bal.Get(25 * time.Millisecond)
```

Comment:
25 ms is a timeout, not a wait time.  If we set it to 0, we would likely fail to get anything back within that time.